csrc/kernel_ir.h

```diff
@@ -166,6 +167,68 @@ class TensorIndex final : public Val {
   Val* index_ = nullptr;
 };
 
+// In theory, we should just put this struct into class Asm, but unfortunately,
+// due to compiler bug, we can not do that:
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88165
+struct AsmOptions {
+  bool volatile_ = false;
+  bool memory = false;
```

Comment:
nit: Why one has a trailing underscore and another doesn't?

Reply:
In principle, for struct, we should not use trailing underscore. But because `volatile` is a C++ keyword, we have to use one to avoid conflict.